src/App.js

```diff
@@ -1,61 +1,133 @@
+import { createPortal } from 'react-dom'
 
-import Board from './components/Board';
-
-const PLAYER_1 = 'X';
-const PLAYER_2 = 'O';
+const PLAYER_1 = 'X'
+const PLAYER_2 = 'O'
 
 const generateSquares = () => {
-  const squares = [];
+  const squares = []
 
-  let currentId = 0;
+  let currentId = 0
 
   for (let row = 0; row < 3; row += 1) {
-    squares.push([]);
+    squares.push([])
     for (let col = 0; col < 3; col += 1) {
       squares[row].push({
         id: currentId,
-        value: '',
-      });
-      currentId += 1;
+        value: ''
+      })
+      currentId += 1
     }
   }
 
-  return squares;
+  return squares
 }
 
 const App = () => {
+  const [squares, setSquares] = useState(generateSquares()); 
+  const [turn, setTurn] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('');
 
-  const [squares, setSquares] = useState(generateSquares());
+  //const
 
   // Wave 2
-  // You will need to create a method to change the square 
+  // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const ChangeSquareValue = square => {
+    let newSquares = [];
 
-  const checkForWinner = () => {
-    // Complete in Wave 3
+    for (let row = 0; row < squares.length; row++) {
+      newSquares.push([]);
+      for (let col = 0; col < squares.length; col++) {
+        if (squares[row][col].id === square && squares[row][col].value === "") {
+          newSquares[row].push({
+            id: squares[row][col].id,
+            value: turn
+          })
 
+          if(turn === PLAYER_1) {
+            setTurn(PLAYER_2)
+          } else {
+            setTurn(PLAYER_1)
+          }
+        } else {
+          newSquares[row].push(squares[row][col]);
+        }
+      }
+    }
+    setSquares(newSquares)
+  }
+ 
+    //complete in wave 3
+  
+  const checkRow = (turn) => {
+    for (let row = 0; row < 3; row++) {
+     if ((squares[row][0].value === turn) &&
+          (squares[row][1].value === turn) &&
+          (squares[row][2].value === turn)){
+            return true;
+          }    
+    }
+    return false;
+  }
+  const checkCol = (turn) => {
+    for (let col = 0; col < 3; col++) {
+     if ((squares[0][col].value === turn) &&
+          (squares[1][col].value === turn) &&
+          (squares[2][col].value === turn)){
+            return true;
+          }    
+    }
+    return false;
   }
+  
+  const checkDiagonal = (turn) => {
+    if ((squares[0][0].value === turn) &&
+          (squares[1][1].value === turn) &&
+          (squares[2][2].value === turn)) {
+            return true;
+          } 
+    if ((squares[0][2].value === turn) &&
+      (squares[1][1].value === turn) &&
+          (squares[2][0].value === turn)){
+            return true;
+          } 
+      return false;
+  }
+  const checkForWinner = (turn) => {
+    if (checkRow(turn) || checkCol(turn) || checkDiagonal(turn)) {
+      setWinner(turn)
+    }
+  }
```

Comment:
I appreciate that you separated this logic into different functions! It definitely helps make it easier to read.